x/concentrated-liquidity/position.go

```diff
@@ -245,6 +244,12 @@ func (k Keeper) deletePosition(ctx sdk.Context,
 	// Remove the position ID to underlying lock ID mapping (if it exists)
 	key = types.KeyPositionIdForLock(positionId)
 	if store.Has(key) {
+		underlyingLockId, err := k.GetLockIdFromPositionId(ctx, positionId)
+		if err != nil {
+			return err
+		}
+		store.Delete(key)
+		key = types.KeyLockIdForPositionId(underlyingLockId)
 		store.Delete(key)
```

Comment:
nit: Reusing the same identifier for various keys is confusing. Suggest changing other instances to be more specific as well

```suggestion
		lockIndexKey := types.KeyLockIdForPositionId(underlyingLockId)
		store.Delete(lockIndexKey)
```

Reply:
https://github.com/osmosis-labs/osmosis/pull/4885/commits/2c6e24f680b0e1f48c2d06a423db51d50bafce17